packetbeat/protos/dns/dns.go

```diff
@@ -501,6 +501,23 @@ func addDNSToMapStr(m common.MapStr, dns *mkdns.Msg, authority bool, additional
 			// etld_plus_one should be removed for 8.0.0.
 			qMapStr["etld_plus_one"] = eTLDPlusOne
 			qMapStr["registered_domain"] = eTLDPlusOne
+
+			if eTLDPlusOne != "" && strings.Contains(eTLDPlusOne, ".") {
+				topLevelDomain := strings.Join(strings.Split(eTLDPlusOne, ".")[1:], ".")
```

Comment:
Splitting and then re-joining seems inefficient. I think it could slice up the eTLDPlusOne string by finding the index of the first dot.